api_spec.yaml

```diff
@@ -1231,11 +1231,217 @@ paths:
+                                          status:
+                                              type: string
+                                              default: "UNKNOWN_TENANT_ERROR"
+                401:
+                    description: Unauthorised access
+                    content:
+                        text/plain:
+                            schema:
+                                type: string
+                                enum:
+                                    - Unauthorised access
+        delete:
+            tags:
+                - Multitenancy
+            summary: Delete the tenant
+            parameters:
+                - name: tenantId
+                  in: query
+                  required: true
+                  schema:
+                      type: string
+                      example: "tenant1"
+                - name: authorization
+                  in: header
+                  required: true
+                  schema:
+                      type: string
+                      example: "Bearer API_KEY"
+            responses:
+                200:
+                    description: Success
+                    content:
+                        application/json:
+                            schema:
+                                type: object
+                                properties:
+                                    status:
+                                        type: string
+                                        default: "OK"
+                                    didExist:
+                                        type: boolean
+                                        default: true
+                401:
+                    description: Unauthorised access
+                    content:
+                        text/plain:
+                            schema:
+                                type: string
+                                enum:
+                                    - Unauthorised access
+        put:
+            tags:
+                - Multitenancy
+            summary: Create or update a tenant
+            parameters:
+                - name: authorization
+                  in: header
+                  required: true
+                  schema:
+                      type: string
+                      example: "Bearer API_KEY"
+            requestBody:
+                content:
+                    application/json:
+                        schema:
+                            type: object
+                            properties:
+                                tenantId:
+                                    type: string
+                                emailPasswordEnabled:
+                                    type: boolean
+                                passwordlessEnabled:
+                                    type: boolean
+                                thirdPartyEnabled:
+                                    type: boolean
```

Comment:
can these be removed and computed from firstFactors array?

Reply:
No this is the update API, the frontend automatically enables or disables them based on the firstFactors

Reply:
I mean can't we compute in the backend itself? it would make the frontend easier. if we are showing toggles only based on firstFactors. think about it